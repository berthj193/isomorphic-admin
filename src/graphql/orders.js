import gql from 'graphql-tag';

export const getList = gql`
query {
  orders {
    id
    customer {
      given_name
      family_name
    }
    grand_total {
      value
      currency
    }
    created_at
  }
}`;

export const getDetails = gql`query ($id: ID!) {
  orders(filter: {
    id: $id
  }) {
    id
    customer {
      given_name
      family_name
      email
    }
    store_key
    grand_total {
      value
      currency
    }
    item_total {
      value
      currency
    }
    shipping_total {
      value
      currency
    }
    item_discount_total {
      value
      currency
    }
    shipping_discount_total {
      value
      currency
    }
    currency
    shipping_address
    shipping_rate
    warehouse
  }
}`;

export const getListFor = gql`query ($auth0Id: String!) {
  orders(filter: {
    customer: {
      auth0Id: $auth0Id
    }
  }) {
    id
    customer {
      given_name
      family_name
      email
    }
    store_key
    grand_total {
      value
      currency
    }
    item_total {
      value
      currency
    }
    shipping_total {
      value
      currency
    }
    item_discount_total {
      value
      currency
    }
    shipping_discount_total {
      value
      currency
    }
    currency
    shipping_address
    shipping_rate
    warehouse
  }
}`;

export const placeOrder = gql`mutation placeOrder(
  $store_key: String!, $shipping_address_id: Int!, $shipping_rate_id: Int!,
  $line_items: String!, $currency: String!, $guest_email: String,
  $guest_given_name: String, $guest_family_name: String, $payment_token: String,
  $discount_code: String, $phone_number: String, $gdpr_optout: Boolean
) {
  placeOrder(
    store_key: $store_key,
    shipping_address_id: $shipping_address_id,
    shipping_rate_id: $shipping_rate_id,
    line_items: $line_items,
    guest_email: $guest_email,
    guest_given_name: $guest_given_name,
    guest_family_name: $guest_family_name,
    currency: $currency,
    payment_token: $payment_token,
    discount_code: $discount_code,
    phone_number: $phone_number,
    gdpr_optout: $gdpr_optout
  ) {
    job_id
    errors
  }
}`;

export const updateOrders = gql`mutation updateOrders(
  $ids: [Int]!
  $store_key: String!, $shipping_address_id: Int!, $shipping_rate_id: Int!,
  $line_items: String!, $currency: String!, $guest_email: String,
  $guest_given_name: String, $guest_family_name: String, $payment_token: String,
  $discount_code: String, $phone_number: String, $gdpr_optout: Boolean
) {
  updateOrders(
    ids: $ids,
    store_key: $store_key,
    shipping_address_id: $shipping_address_id,
    shipping_rate_id: $shipping_rate_id,
    line_items: $line_items,
    guest_email: $guest_email,
    guest_given_name: $guest_given_name,
    guest_family_name: $guest_family_name,
    currency: $currency,
    payment_token: $payment_token,
    discount_code: $discount_code,
    phone_number: $phone_number,
    gdpr_optout: $gdpr_optout
  ) {
    job_id
    errors
  }
}`;

export const createRefund = gql`mutation createRefund(
  $payment_id: ID!, $amount_pence: Int!
) {
  createRefund(
    payment_id: $payment_id,
    amount_pence: $amount_pence
) {
    id
  }
}`;
