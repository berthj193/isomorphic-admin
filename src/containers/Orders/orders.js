import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ItemList from '../../components/list/item-list';
import withOrdersManager from '../../components/apolloProviders/withOrdersManager';
import { toListItem } from '../../mappers/order';

import columns from './columns';
import { getList } from '../../graphql/orders';

export class Orders extends Component {
  static propTypes = {
    orders: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
    mutate: PropTypes.func.isRequired,
  };

  dataSource() {
    return toListItem(this.props.orders);
  }

  render() {
    return (
      <ItemList
        data={this.dataSource()}
        error={this.props.error}
        title="Orders"
        columns={columns}
        buttons={[{
          title: 'New form',
          link: '/dashboard/orders/new',
        }]}
        loading={!this.props.orders && !this.props.error}
      />
    );
  }
}

export default withOrdersManager(Orders, { query: getList });
