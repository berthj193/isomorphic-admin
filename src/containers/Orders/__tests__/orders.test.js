import React from 'react';
import { Orders } from '../orders';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import OrderClient from '../../../helpers/client/orderManagerClient';

jest.mock('../../../helpers/client/orderManagerClient');

describe('<Orders />', () => {
  const mutate = jest.fn();
  const defaultProps = { mutate };
  const getComponent = (props = {}) => shallow(<Orders {...Object.assign(defaultProps, props)} />);
  const mockOrders = [
    {
      id: '1',
      customer: {
        given_name: 'John',
        family_name: 'Doe',
        email: 'johndoe@isomorphic.com',
      },
      grand_total: {
        value: 0,
        currency: 'GBP',
      },
      created_at: '2018-05-31 15:43:15 UTC',
    },
  ];

  beforeEach(() => {
    jest.resetAllMocks();
    OrderClient.query.mockReturnValue(Promise.resolve({ data: { stores: [] } }));

    mutate.mockReturnValue(Promise.resolve({ foo: 'bar' }));
  });

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });

  it('should map and pass orders to table component', () => {
    const component = getComponent();
    component.instance().setState({
      orders: mockOrders,
    });

    expect(toJson(component.update())).toMatchSnapshot();
  });

  it('dataSource method should return mapped object', () => {
    const component = getComponent({ orders: mockOrders });
    expect(component.instance().dataSource().length).toBe(1);
  });
});
