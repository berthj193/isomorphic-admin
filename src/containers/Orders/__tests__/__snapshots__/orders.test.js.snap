// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Orders /> should map and pass orders to table component 1`] = `
<withRouter(ItemList)
  buttons={
    Array [
      Object {
        "link": "/dashboard/orders/new",
        "title": "New form",
      },
    ]
  }
  columns={
    Array [
      Object {
        "dataIndex": "id",
        "key": "id",
        "title": "ID",
      },
      Object {
        "dataIndex": "customer_name",
        "key": "customer_name",
        "title": "Name",
      },
      Object {
        "dataIndex": "email",
        "key": "email",
        "title": "Email",
      },
      Object {
        "dataIndex": "created_at",
        "key": "created_at",
        "title": "Ordered At",
      },
      Object {
        "dataIndex": "status",
        "key": "status",
        "title": "Status",
      },
      Object {
        "dataIndex": "grand_total",
        "key": "grand_total",
        "title": "Order Total",
      },
    ]
  }
  data={Array []}
  loading={true}
  title="Orders"
/>
`;

exports[`<Orders /> should render properly 1`] = `
<withRouter(ItemList)
  buttons={
    Array [
      Object {
        "link": "/dashboard/orders/new",
        "title": "New form",
      },
    ]
  }
  columns={
    Array [
      Object {
        "dataIndex": "id",
        "key": "id",
        "title": "ID",
      },
      Object {
        "dataIndex": "customer_name",
        "key": "customer_name",
        "title": "Name",
      },
      Object {
        "dataIndex": "email",
        "key": "email",
        "title": "Email",
      },
      Object {
        "dataIndex": "created_at",
        "key": "created_at",
        "title": "Ordered At",
      },
      Object {
        "dataIndex": "status",
        "key": "status",
        "title": "Status",
      },
      Object {
        "dataIndex": "grand_total",
        "key": "grand_total",
        "title": "Order Total",
      },
    ]
  }
  data={Array []}
  loading={true}
  title="Orders"
/>
`;
